### PR TITLE
Use cached chart data for place explorer

### PR DESCRIPTION
**Pull request by beets**
- keeps only latest data unless line charts are required
- client still fills in with stats data if data isn't cached yet

### CODE REVIEW

**Comment by beets** [comment]
latest screenshot: https://staging.datacommons.org/dev/screenshot/ac111bc

ptal when you've the chance, thanks!

File: static/js/shared/data_fetcher.ts [hunk 1]
```diff
@@ -19,17 +19,25 @@ import axios, { AxiosResponse } from "axios";
...
   };
   placeName: string;
   placeDcid: string;
   provenanceDomain: string;
 }
 
-interface ApiResponse {
-  [key: string]: TimeSeries | null;
+interface StatApiResponse {
+  [place_dcid: string]: TimeSeries | null;
```
Comment (shifucun) [inline static/js/shared/data_fetcher.ts:34]: should this be placeDcid?

File: server/routes/api/chart.py [hunk 2]
```diff
@@ -65,13 +65,57 @@ def build_url(dcid, stats_vars):
...
+
+    Args:
+        timeseries: GetStats timeseries, dictionary of date: value
+
+    Returns:
+        GetStats timeseries with only one date/value, or None
+    """
+    if not timeseries: return None
+    max_date = max(timeseries)
+    filtered_series = {}
```
[the remark this answers is not in the document]
Comment (beets) [reply server/routes/api/chart.py:93]: oh nice, that's much better, thanks!

File: server/routes/api/chart.py [hunk 3]
```diff
@@ -65,13 +65,57 @@ def build_url(dcid, stats_vars):
     return urllib.parse.unquote(url_for('tools.timeline', _anchor=anchor))
 
 
+def get_statsvars_need_all_dates(chart_list, sv_set):
```
[the remark this answers is not in the document]
Comment (beets) [reply server/routes/api/chart.py:68]: ok, removed sv_set, and returns a new set instead that the caller can combine.

File: server/routes/api/chart.py [hunk 4]
```diff
@@ -92,16 +136,27 @@ def config(dcid):
...
-            used_stats_vars.update(set(chart['statsVars']))
-        for child in section['children']:
-            for chart in child['charts']:
-                used_stats_vars.update(set(chart['statsVars']))
-
-    # Population the GNI url to each chart.
+    # Track stats vars where we need data from all dates (i.e. line charts)
+    sv_keep_all_dates = set()
+    for topic in cc:
+        get_statsvars_need_all_dates(topic['charts'], sv_keep_all_dates)
```
[the remark this answers is not in the document]
Comment (beets) [reply server/routes/api/chart.py:142]: done

File: static/js/shared/data_fetcher.ts (hunk 1 above)
Comment (beets) [reply static/js/shared/data_fetcher.ts]: done

**Comment by beets** [comment]
thanks for the review! addressed your comments

**Pull request merged**